File: Begasp.net VB 9781118846773_VB_Full_Download/Source/Chapter08/Controls/Banner.ascx.vb
Partial Class Controls_Banner
  Inherits System.Web.UI.UserControl

  Public Property DisplayDirection As Direction

  Public Property NavigateUrl() As String
    Get
      Dim _navigateUrl As Object = ViewState("NavigateUrl")
      If _navigateUrl IsNot Nothing Then
        Return CType(_navigateUrl, String)
      Else
        Return "http://p2p.wrox.com" ' Return a default value
      End If
    End Get
    Set(Value As String)
      ViewState("NavigateUrl") = Value
    End Set
  End Property

  Protected Sub Page_Load(sender As Object, e As EventArgs) Handles Me.Load
    Select Case DisplayDirection
      Case Direction.Horizontal
        HorizontalPanel.Visible = True
        VerticalPanel.Visible = False
        HorizontalLink.HRef = NavigateUrl
      Case Direction.Vertical
        VerticalPanel.Visible = True
        HorizontalPanel.Visible = False
        VerticalLink.HRef = NavigateUrl
    End Select
  End Sub
End Class

File: TobeDemo/Controls/Banner.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace TobeDemo.Controls
{
    public partial class Banner : System.Web.UI.UserControl
    {
        protected void Page_Load(object sender, EventArgs e)
        {
            switch (displayDirection)
            {
                case Direction.horizontal:
                    HorizontalPanel.Visible = true;
                    VerticalPanel.Visible = false;
                    HorizontalLink.HRef = NavigateUrl;
                    break;
                case Direction.vertical:
                    HorizontalPanel.Visible = false;
                    VerticalPanel.Visible = true;
                    VerticalLink.HRef = NavigateUrl;
                    break;
                default:
                    HorizontalPanel.Visible = true;
                    VerticalPanel.Visible = true;
                    HorizontalLink.HRef = NavigateUrl;
                    VerticalLink.HRef = NavigateUrl;
                    break;
            }
        }

        public Direction displayDirection
        { 
            get
            {
                object _dir=ViewState["displayDirection"];
                if (_dir != null) return (Direction)_dir;
                else return Direction.horizontal;
            }
            set { ViewState["displayDirection"] = value; this.Page_Load(new object() , new EventArgs()); }
        }

        public string NavigateUrl
        {
            get
            {
                object _nav=ViewState["NavigateUrl"];
                if (_nav != null) return (string)_nav;
                else return "http://www.google.com";
            }
            set { ViewState["NavigateUrl"] = value; this.Page_Load(new object(), new EventArgs()); }
        }
    }
}